include/swift/Frontend/InputFile.h

```diff
@@ -41,7 +41,9 @@ class InputFile {
   /// string.
   InputFile(StringRef name, bool isPrimary,
             llvm::MemoryBuffer *buffer = nullptr)
-      : Filename(name), IsPrimary(isPrimary), Buffer(buffer) {
+      : Filename(
+            convertBufferNameFromLLVM_getFileOrSTDIN_toSwiftConventions(name)),
```

Comment:
I don't think I actually like this here. It seems weird that InputFile should have to know that there are two different conventions. That seems like the caller's perogative.

Reply:
I understand, but on the other hand there are four callers (one in CompilerInvocation and three in SwiftDocSupport) that clearly need the conversion and possibly others. If in the future, someone adds a call to this function, it would not be surprising to leave the conversion out.
Seems more robust to leave it this way. Does the weirdness dominate the fragility?

Reply:
I don't think the ones in SwiftDocSupport are likely to encounter stdin as an input file, but all right.

Reply:
Thanks.